tls/tls_srv.c

```diff
@@ -912,7 +989,13 @@ ttls_parse_client_hello(TlsCtx *tls, unsigned char *buf, size_t len,
 		}
 		io->hslen -= 2;
 		n = tls->hs->ext_sz;
-		if (n + 4 > tls->hs->ext_rem_sz || n > sizeof(tls->hs->ext)) {
+                /*
+                 * Do not account size of @ext for unsupportred extensions.
+                 * They must not be copied to it.
+                 */
+		if (n + 4 > tls->hs->ext_rem_sz
+                    || (ttls_ext_is_supported(tls->hs->ext_type)
+                        && n > sizeof(tls->hs->ext))) {
```

Comment:
The Linux kernel uses tabs for alignment with not more than 7 spaces at the end for alignment. We also usually place `{` on a new line after multi-line conditions